pyccel/ast/core.py

```diff
@@ -5177,15 +5177,56 @@ def bodies(self):
         return b
 
 
-class IfTernaryOperator(If):
+class IfTernaryOperator(Basic, PyccelAstNode):
+    """Represents a IfTernaryOperator statement in the code.
 
-    """class for the Ternery operator"""
+    Parameters
+    ----------
+    args :
+        args : type list
+        format : [condition , value_if_true, value_if_false]
+
+    Examples
+    --------
+    >>> from sympy import Symbol
+    >>> from pyccel.ast.core import Assign, IfTernaryOperator
+    >>> n = Symbol('n')
+    >>> x = 5 if n > 1 else 2
+    >>> IfTernaryOperator([PyccelGt(n > 1),  5,  2])
+    IfTernaryOperator([PyccelGt(n > 1),  5,  2])
+    """
     def __init__(self, *args):
```

Comment:
Can an `IfTernaryOperator` be initialised with an arbitrary number of arguments? I would have thought that it would require exactly 3

Reply:
normally it should be 3 arguments for the format   _value_if_cond__true if _cond_ else _value_cond_false_

Reply:
In that case I think it would be better if we have:
`def __init__(self, cond, value_true, value_false):`
Currently if someone misunderstands how to use the `IfTernaryOperator` then they might provide extra args and not realise that they are unused. For example if they wanted to construct a case like:
`a if b else c if d else e`
then allowing a variable number of arguments might make them believe that they can do `IfTernaryOperator([b, a, d, c, e])`
It also means that we have to read the function body in order to know which arguments need to be passed